### DISCUSSION TITLE
Telegram Friendly Embeds

### DISCUSSION THREAD

**Discussion by LeviSnoot** [diced/zipline#452] [Ideas]
Currently, embeds work really well on discord, but less so on Telegram.

For example, Telegram does not respect the **Embed Title** field on the Manage User page. It appears to be showing the filename on that field instead.

The second thing with Telegram embeds in zipline is they do not embed videos. Discord does this fine, and I know Telegram has the ability to do so. It just doesn't unfortunately.

**Comment by diced** [comment]
https://github.com/diced/zipline/discussions/453

Similar to this, I'm not really sure what I can do to get this working.

**Reply by LeviSnoot** [reply]
As I'm taking a look at the meta tags I believe telegram is grabbing either the `twitter:title` tag or the HTML title tag itself. Although I find the latter unlikely.

Here's the meta tags for an upload to my instance:

![](https://github.com/diced/zipline/assets/13402525/5d1e190b-9b8f-4abf-aacc-29c9a887c6be)

and here's what telegram provides when posting the link:

![](https://github.com/diced/zipline/assets/13402525/d04cdb20-c9d5-4d9a-b2bb-82fa5de72315)

My Embed title is: `{file.originalName}`, so in this instance the text underneath "Levi's Bucket" should be `originalName.png`, but instead it's `fm5aCX.png`. It'd be nice to have a field for "Page Title", and/or just grabbing the same value as "Embed Title" for `twitter:title`, `og:title` and `<title>`.

As for videos I'm as lost as you are, all the tags are there as far as I can tell. That being said, the "video thumbnails" mentioned in the docs never seem to generate for me either, if that part worked it wouldn't be as much of an issue for me.
